### PR TITLE
Postgresql parameters for db query investigations

### LINKED ISSUE #4318: Explore database configurations to help with investigations and tuning query usage and performance
1. Enable a long query log: Set the Postgres configuration with something like this log_min_duration_statement = 2000
2. It might be worth enabling auto_explain, which we can use to find unstable execution plans.
3. Enable pg_stat_statements. We can use this to find the queries that add the most load. Long-running queries are a problem, but efficient queries we run often can also add substantial load. We can use information from this to find out how much total execution time the database spends on all our queries.

#4064

### PR DESCRIPTION
**Pull request by alismx**
# DEVOPS/DATABASE PULL REQUEST

## Related Issue

- resolves #4318

## Changes Proposed

- Installs `pg_stat_statement` extension ( this will be installed manually )
- Sets up terraform configuration for the following PostgreSQL parameters
  - `log_min_duration_statement`
  - `track_io_timing`
  - `pg_stat_statements.track`
  - `auto_explain.log_analyze`
  - `auto_explain.log_min_duration`
- Adds `auto_explain` to our databases list of `shared_preload_libraries` (previously was `pg_cron,pg_stat_statements`)
- Limits our `azure.extensions` to extensions we use. This was manually set up and had way more than we needed or used.

## Additional Information

- Manual rollout steps:
  - Create a db backup
  - install `pg_stat_statements`
  - All databases will need to be restarted to add `auto_explain` to our list of `shared_preload_libraries`
  - merge this pr
  - Smoke test environment
  - delete db backup

- I tried to add the `pg_stat_statements` extension with a backend migration, but our migrations user doesn't have permission
- All these parameters have been tested to work but have been disabled because these options can cause performance issues, so we only want to enable them when we believe we need them.
- This is work to enable future devs to investigate query issues.
- I'd like to run an exercise using these parameters to gather baseline data about our queries.

Wiki Page: https://github.com/CDCgov/prime-simplereport/wiki/Database-investigation-tools

## Testing

- How should reviewers verify this PR?

- [ ] Check the terraform plan.

### Database
- [ ] Only database changes are included in this PR
- [ ] Any new tables or columns that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
- [ ] Any changes to tables that have custom no-PHI views are accompanied by changes to those views (including re-granting permission to the no-PHI user if need be)
- [ ] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [ ] Rollback has been verifed locally and in a deployed environment
- [ ] Any changes to the startup configuration have been documented in the README

### Infrastructure
- [ ] Consult the results of the `terraform-plan` job inside the "Terraform Checks" workflow run for this PR. Confirm that there are no unexpected changes!

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

### Cloud
- [ ] Oncall has been notified if this change is going in after-hours
- [ ] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### Documentation
- [ ] Any changes to the startup configuration have been documented in the README

### CODE REVIEW

File: ops/services/postgres_db/investigation.tf [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# These are disabled by default to avoid performance impact.
+
+# https://www.postgresql.org/docs/current/runtime-config-logging.html
+resource "azurerm_postgresql_flexible_server_configuration" "log_min_duration_statement" {
+  name      = "log_min_duration_statement"
+  server_id = azurerm_postgresql_flexible_server.db.id
+  # To enable: this is in milliseconds, update this to a positive value in milliseconds to enable this logging
```
Comment (emyl3) [inline ops/services/postgres_db/investigation.tf:7]: Question for my own learning: is there a good value to start off with when turning this on? Is every minute too much too little? 🤔

**Review by emyl3 — APPROVED** [review]
Left one non-blocking question! Thank you for adding docs on how to use this 😲

**Review by BobanL — APPROVED** [review]
Changes look good! Thanks for adding the documentation 😄

File: ops/services/postgres_db/investigation.tf (hunk 1 above)
Comment (alismx) [reply ops/services/postgres_db/investigation.tf:7]: The answer to this depends very much on the context, but I expect most queries to complete well under 1 second, so in most cases, setting it to something between 2 and 5 seconds would probably be fine.

**Comment by rin-skylight** [comment]
Quick note: Let's make sure we take a backup of the `prod` DB before we roll this out. Just in case!

**Review by rin-skylight — APPROVED** [review]

**Comment by alismx** [comment]
I'll deploy this and perform the manual DB updates on the week of July 26th.

**Pull request merged**